### DISCUSSION TITLE
Strong name key

### DISCUSSION THREAD

**Discussion by twtb** [SteveGilham/altcover#134] [Q&A]
Could somebody please provide an example on how to use the -k and --sn options, to change the strong name key of the instrumented examples?
What's the difference between the two? And what do they expect as argument, a file?

**Comment by SteveGilham** [comment]
The difference between them

`--sn` says "take this key, and strong-name every instrumented assembly with it, whether or not strong-named on input"
`-k` says "here are one or more strong-name keys; if the assembly you are instrumenting was strong-named by one of them, use it so the instrumented assembly name matches the input"

They take the key-file as argument(s), the one that you would use in the build, like

![Screenshot 2021-11-30 091607](https://user-images.githubusercontent.com/7460854/144019222-12aafe4f-cfb0-4f97-8e72-b5ab79fe8902.png)

and it gets used like in this excerpt from the AltCover build process
```
Starting task 'AltCover'
/usr/bin/mono --debug /home/runner/work/altcover/altcover/_Binaries/AltCover/Release+AnyCPU/net472/AltCover.exe -o ./__UnitTestWithAltCoverRunner -s "\.DataCollector" -s Sample -t "System\." -t "Sample3\.Class2" -t Microsoft -t ICSharpCode -t UnitTestStub -t SolutionRoot -m WaitForExitCustom --sn /home/runner/work/altcover/altcover/Build/SelfTest.snk --reportFormat OpenCover -r /home/runner/work/altcover/altcover/_Reports/UnitTestWithAltCoverRunner.xml --single --localSource
```

The motivation is primarily for .net Framework contexts, where you can get issues with the assembly name changing in the instrumentation process -- something built can take an assembly-qualified reference to one of the types in a dependency, and if the instrumented assembly is not strong-names as the original was, that type is no longer there when running the instrumented tests -- examples that come to mind are with [NUnit](https://github.com/coverlet-coverage/coverlet/issues/525) and some uses of [F# quotations](https://github.com/SteveGilham/altcover/commit/b1c67e15f5fd7f89f72d49d3e73dd960c6917b6e)
